tools/setup_helpers/cmake.py

```diff
@@ -287,6 +285,14 @@ def generate(
             elif var.startswith(('BUILD_', 'USE_', 'CMAKE_')):
                 build_options[var] = val
 
+        # The default value cannot be easily obtained in CMakeLists.txt. We set it here.
+        py_lib_path = cast(str, sysconfig.get_path('purelib'))
+        cmake_prefix_path = cast(str, build_options.get('CMAKE_PREFIX_PATH', None))
+        if cmake_prefix_path:
+            build_options["CMAKE_PREFIX_PATH"] = f'{py_lib_path};{cmake_prefix_path}'
+        else:
+            build_options['CMAKE_PREFIX_PATH'] = py_lib_path
```

Comment:
nit: this can simplify right? 
```suggestion
        cmake_prefix_path = cast(str, build_options.get('CMAKE_PREFIX_PATH', ''))
        build_options["CMAKE_PREFIX_PATH"] = f'{py_lib_path};{cmake_prefix_path}'
```

Reply:
Yes, technically they are equivalent. I just wanted the final value of `CMAKE_PREFIX_PATH` to be more "pedantic". For someone checking out CMakeCache.txt later, seeing a `CMAKE_PREFIX_PATH` that ends with a semicolon might give suspicion about whether it was intentional or whether there was a bug in the build system that overwrote/skipped something.